pkg/controller/machineset/controller.go

```diff
@@ -34,14 +35,18 @@ import (
 	clusterapiclientset "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
 	listers "sigs.k8s.io/cluster-api/pkg/client/listers_generated/cluster/v1alpha1"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
+	"sigs.k8s.io/cluster-api/util"
 )
 
 // controllerKind contains the schema.GroupVersionKind for this controller type.
 var controllerKind = v1alpha1.SchemeGroupVersion.WithKind("MachineSet")
 
-// reconcileMutexSleepSec is the duration to sleep before releasing the mutex lock that is held for reconcilation.
-// See https://github.com/kubernetes-sigs/cluster-api/issues/245
-var reconcileMutexSleepSec = time.Second
+// stateConfirmationTimeout is the amount of time allowed to wait for desired state.
+var stateConfirmationTimeout = 10 * time.Second
+
+// stateConfirmationInterval is the amount of time between polling for the desired state.
+// The polling is against a local memory cache.
+var stateConfirmationInterval = 100 * time.Millisecond
```

Comment:
Is this too fast?

Reply:
this is against an local internal cache, it shouldnt be an issue. 
i can raise the interval if it is a concern